java-manta-client/src/main/java/com/joyent/manta/util/MantaUtils.java

```diff
@@ -373,6 +376,44 @@ public static String lastItemInPath(final String path) {
         return lastPart.toString();
     }
 
+    /**
+     * Compute the intermediate directories between "/" and {@code rawPath}.
+     *
+     * @param rawPath The fully qualified path of the Manta directory.
+     * @return All non-root directories leading to and including rawPath
+     */
+    public static String[] prefixPaths(final String rawPath) {
+        final String[] parts = rawPath.split(SEPARATOR);
```

Comment:
We may want to skip using `String.split()` here and use `StringUtils.split()` because we don't need regular expression parsing. However, things have changed enough in the JDK, that this may be just superstition and old habits on my part.

EDIT: I'm not the only one confused: https://stackoverflow.com/a/11002374/33611

Reply:
Since `SEPARATOR` is a string I ended up with `StringUtils.split(rawPath, SEPARATOR.charAt(0))`. The output is different as well, previously I'd get an empty string as the first element since the string started with `/`. One less unneeded loop.